### PR TITLE
Simplify streaming_merge function parameters

### PR DESCRIPTION
**Pull request by mertak-synnada**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Simplification of streaming_merge function parameters, since they're too many (7 currently).

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/physical-plan/src/sorts/sort.rs [hunk 1]
```diff
@@ -342,15 +342,15 @@ impl ExternalSorter {
...
-                self.fetch,
-                self.reservation.new_empty(),
-            )
+                schema: Arc::clone(&self.schema),
+                expressions: &self.expr,
+                metrics: self.metrics.baseline.clone(),
+                batch_size: self.batch_size,
+                fetch: self.fetch,
+                reservation: self.reservation.new_empty(),
+            })
```
Comment (alamb) [inline datafusion/physical-plan/src/sorts/sort.rs:353]: Another potential API to consider might be a builder style

```suggestion
            StreamingMergeBuilder::new()
              .with_streams(streams)
              .with_schema(Arc::clone(&self.schema))
              .with_expressions(&self.expr)
              .with_metrics(self.metrics.baseline.clone())
              .with_batch_size(self.batch_size)
              .with_fetch(self.fetch)
              .with_reservation(self.reservation.new_empty())
              .build()?
```

**Review by alamb — APPROVED** [review]
Thank you @mertak-synnada  -- this looks like an improvement over the current code to me ❤️

File: datafusion/physical-plan/src/sorts/sort.rs (hunk 1 above)
Comment (comphead) [reply datafusion/physical-plan/src/sorts/sort.rs:353]: 👍 for the builder

Comment (mertak-synnada) [reply datafusion/physical-plan/src/sorts/sort.rs]: Thank you for the review and suggestion! I've implemented the builder pattern like you suggested.

**Review by alamb — APPROVED** [review]
Love it -- thank you @mertak-synnada  and @comphead  ❤️

**Pull request merged**